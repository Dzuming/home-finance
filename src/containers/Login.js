import React, { Component } from 'react';
import compose from 'recompose/compose';
import InputText from '../components/InputText';
import Grid from 'material-ui/Grid';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as loginActions from '../actions/loginActions';
import { withStyles } from 'material-ui/styles';
import RaisedButton from '../components/RaisedButton';

const styles = () => ({
  container: {
    width: '100vw',
    height: '100vh'
  }
});

class Login extends Component {
  state = {
    login: '',
    password: ''
  };
  sendCredentials = () => {
    this
      .props.actions.login({email: this.state.login, password: this.state.password});
  };
  handleChange = name => event => {
    this.setState({[name]: event.target.value});
  };

  render () {
    const classes = this.props.classes;
    const {login, password} = this.state;
    return (
      <div>
        <Grid
          container
          className={classes.container}
          alignItems='center'
          direction='row'
          justify='center'>
          <form noValidate autoComplete="off">
            <div>
              <InputText
                type="email"
                label="Login"
                value={login}
                changeMethod={this.handleChange('login')}/>
            </div>
            <div>
              <InputText
                type="password"
                label="Password"
                value={password}
                changeMethod={this.handleChange('password')}/>
            </div>
            <RaisedButton
              clickMethod={this.sendCredentials}
              text={'Login'}
              color={'default'}/>
          </form>
        </Grid>
      </div>
    );
  }
}

Login.propTypes = {
  classes: PropTypes.object.isRequired,
  actions: PropTypes.object,
  history: PropTypes.object,
  isAuthenticated: PropTypes.bool.isRequired
};

function mapStateToProps (state) {
  return {isAuthenticated: state.auth.isAuthenticated};
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(loginActions, dispatch)
  };
}

export default compose(withStyles(styles), connect(mapStateToProps, mapDispatchToProps))(Login);
