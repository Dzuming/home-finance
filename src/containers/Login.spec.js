import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Login from './Login';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  token: '',
  user: [],
  login: { isAuthenticated: true },
});
const component = create(
  <Provider store={store}>
    <Login />
  </Provider>,
);

describe('<Login />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
