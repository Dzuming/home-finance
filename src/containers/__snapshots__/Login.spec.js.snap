// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
<div>
  <div
    className="MuiGrid-typeContainer-2 MuiGrid-spacing-xs-16-24 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-center-19 Connect-Login--container-1"
  >
    <form
      autoComplete="off"
      noValidate={true}
    >
      <div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-94 InputText-textField-93"
          onBlur={[Function]}
          onFocus={[Function]}
          required={false}
        >
          <label
            className="MuiFormLabel-root-104 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102"
            data-shrink={false}
            htmlFor={undefined}
          >
            Login
          </label>
          <div
            className="MuiInput-root-108 MuiInput-formControl-109 MuiInput-inkbar-110 MuiInput-underline-114"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-117 MuiInput-inputSingleline-120"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="email"
              value={undefined}
            />
          </div>
        </div>
      </div>
      <div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-94 InputText-textField-93"
          onBlur={[Function]}
          onFocus={[Function]}
          required={false}
        >
          <label
            className="MuiFormLabel-root-104 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102"
            data-shrink={false}
            htmlFor={undefined}
          >
            Password
          </label>
          <div
            className="MuiInput-root-108 MuiInput-formControl-109 MuiInput-inkbar-110 MuiInput-underline-114"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-117 MuiInput-inputSingleline-120"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="password"
              value={undefined}
            />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root-139 MuiButton-root-125 MuiButton-raised-131 RaisedButton-button-123"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-127"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-141"
        />
      </button>
    </form>
  </div>
</div>
`;
